propagator/opentelemetry-propagator-jaeger/src/opentelemetry/propagators/jaeger/__init__.py

```diff
@@ -81,7 +81,9 @@ def inject(
         if span_context == trace.INVALID_SPAN_CONTEXT:
             return
 
-        span_parent_id = span.parent.span_id if span.parent else 0
+        span_parent_id = (
```

Comment:
Could it ever be not recording but still has parent id?

Reply:
Possibly? But same thing could be said with `span.kind` bug that we found in the instrumentations, but we remedied that by assuming it could only happen with `NonRecordingSpan`. We should at least be consistent.